From d84986c9c4e030eb54c78ae6b6803f9329003bc1 Mon Sep 17 00:00:00 2001
From: Nikolay Aleksandrov <nikolay@redhat.com>
Date: Fri, 5 Sep 2014 11:04:08 -0400
Subject: [ethernet] sfc: Remove DEFINE_PCI_DEVICE_TABLE macro use

Message-id: <1409915048-26438-33-git-send-email-nikolay@redhat.com>
Patchwork-id: 91429
O-Subject: [RHEL7.1 PATCH BZ 1110888 32/32] PCI: Remove DEFINE_PCI_DEVICE_TABLE macro use
Bugzilla: 1110888
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

sfc part of upstream commit:
commit 9baa3c34ac4e27f7e062f266f50cc5dbea26a6c1
Author: Benoit Taine <benoit.taine@lip6.fr>
Date:   Fri Aug 8 15:56:03 2014 +0200

    PCI: Remove DEFINE_PCI_DEVICE_TABLE macro use

    We should prefer `struct pci_device_id` over
`DEFINE_PCI_DEVICE_TABLE` to
    meet kernel coding style guidelines.  This issue was reported by
checkpatch.

    A simplified version of the semantic patch that makes this change is
as
    follows (http://coccinelle.lip6.fr/):

    // <smpl>

    @@
    identifier i;
    declarer name DEFINE_PCI_DEVICE_TABLE;
    initializer z;
    @@

    - DEFINE_PCI_DEVICE_TABLE(i)
    + const struct pci_device_id i[]
    = z;

    // </smpl>

    [bhelgaas: add semantic patch]
    Signed-off-by: Benoit Taine <benoit.taine@lip6.fr>
    Signed-off-by: Bjorn Helgaas <bhelgaas@google.com>

Signed-off-by: Nikolay Aleksandrov <nikolay@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/sfc/efx.c b/drivers/net/ethernet/sfc/efx.c
index d257270..3bc077f 100644
--- a/drivers/net/ethernet/sfc/efx.c
+++ b/drivers/net/ethernet/sfc/efx.c
@@ -2642,7 +2642,7 @@ void efx_schedule_reset(struct efx_nic *efx, enum reset_type type)
  **************************************************************************/
 
 /* PCI device ID table */
-static DEFINE_PCI_DEVICE_TABLE(efx_pci_table) = {
+static const struct pci_device_id efx_pci_table[] = {
  {PCI_DEVICE(PCI_VENDOR_ID_SOLARFLARE,
       PCI_DEVICE_ID_SOLARFLARE_SFC4000A_0),
   .driver_data = (unsigned long) &falcon_a1_nic_type},
-- 
1.7.1